diff --git a/src/MultiDimFit.cc b/src/MultiDimFit.cc
index f7f5925c..08260254 100644
--- a/src/MultiDimFit.cc
+++ b/src/MultiDimFit.cc
@@ -978,6 +978,7 @@ void MultiDimFit::doRandomPoints(RooWorkspace *w, RooAbsReal &nll)
             if (ok) {
                 double qN = 2*(nll.getVal() - nll0);
                 double prob = ROOT::Math::chisquared_cdf_c(qN, n+nOtherFloatingPoi_);
+                deltaNLL_ = nll.getVal() - nll0;
 		for(unsigned int j=0; j<specifiedNuis_.size(); j++){
 			specifiedVals_[j]=specifiedVars_[j]->getVal();
 		}
