diff --git a/CombineTools/python/combine/CombineToolBase.py b/CombineTools/python/combine/CombineToolBase.py
index 64c0da7..6abeef4 100755
--- a/CombineTools/python/combine/CombineToolBase.py
+++ b/CombineTools/python/combine/CombineToolBase.py
@@ -5,6 +5,8 @@ import stat
 from functools import partial
 from multiprocessing import Pool
 from six.moves import range
+import re
+import random
 
 DRY_RUN = False
 
@@ -348,6 +350,17 @@ class CombineToolBase:
                 for line in self.job_queue[j:j + self.merge]:
                     newline = line
                     if line.startswith('combine'): newline = self.pre_cmd + line.replace('combine', './combine', 1)
+                    # Are we asking for a random seed?
+                    if re.search('-s +-1', newline):
+                        # Generate a new random seed
+                        rseed = random.randint(1000,1000000)
+                        # Update the seed on the combine call
+                        newline = re.sub('-s +-1', '-s {}'.format(rseed), newline)
+                        points = re.findall('--points \d*', newline)[0].split()[-1]
+                        fpoint = re.findall('--firstPoint \d*', newline)[0].split()[-1]
+                        lpoint = re.findall('--lastPoint \d*', newline)[0].split()[-1]
+                        split = int(lpoint) - int(fpoint) + 1
+                        newline = re.sub('--points \d*', '--points {}'.format(split), newline)
                     wsp = str(self.extract_workspace_arg(newline.split()))
 
                     newline = newline.replace(wsp, os.path.basename(wsp))
